### PR TITLE
firefoxpwa: add fhs wrapper to run downloaded binaries

### LINKED ISSUE #302431: firefoxpwa : No such file or directory (os error 2)
### Describe the bug
After installing firefoxpwa and installing one PWA, when I try to launch it, I get this error : `No such file or directory (os error 2)`

### Steps To Reproduce
1. Install firefoxpwa this way in /etc/nixos/configuration.nix :
```
{ config, pkgs, lib, inputs, ... }:

{
  environment.systemPackages = with pkgs; [
    inputs.nixpkgs_unstable.legacyPackages."${pkgs.system}".firefoxpwa
  ];
  programs.firefox = {
    enable = true;
    package = pkgs.firefox;
    nativeMessagingHosts.packages = with pkgs; [
      browserpass
      inputs.nixpkgs_unstable.legacyPackages."${pkgs.system}".firefoxpwa
    ];
  };
```
2. Install firefox extension https://addons.mozilla.org/fr/firefox/addon/pwas-for-firefox/
3. be sure this folder exists : ~/.local/share/icons/hicolor/ to solve this issue : https://github.com/NixOS/nixpkgs/issues/301194
4. Install a PWA
5. Launch the PWA
6. Get the following error : `No such file or directory (os error 2)`

### Expected behavior
The PWA should launch

### Screenshots
image

### Additional context
flake.nix file : 
```
{
  description = "Thatoo's flake";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
    nixpkgs_unstable.url = "github:NixOS/nixpkgs/nixos-unstable";
  };

  outputs = inputs@{ self, nixpkgs, ... }: {

    nixosConfigurations.default = nixpkgs.lib.nixosSystem {
      system = "x86_64-linux";
      specialArgs = { inherit inputs; };
      modules = [
        ./configuration.nix
      ];
    };
  };
}
```

### Notify maintainers

@camillemndn
@pasqui23

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.84, NixOS, 23.11 (Tapir), 23.11.20240406.e38d7cb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pasqui23**
## Description of changes
Fixes https://github.com/NixOS/nixpkgs/issues/302431?notification_referrer_id=NT_kwDOAGnFH7MxMDE2Mzg4NjI2Mzo2OTMxNzQz#issuecomment-2053956619

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pasqui23** [comment]
What does " This PR does not cleanly list package outputs after merging" mean?

**Comment by tiecia** [comment]
Would this PR potentially fix this issue that I am having?

https://github.com/NixOS/nixpkgs/issues/306167

File: pkgs/by-name/fi/firefoxpwa/package.nix [hunk 1]
```diff
@@ -16,84 +16,111 @@
...
+  fhs = buildFHSUserEnv {
+    name = "${name}-wrapper-${version}";
+    runScript = "${unwrapped}/bin/.firefoxpwa-wrapped";
+    targetPkgs = pkgs: [
+      pkgs.dbus-glib
+      pkgs.gtk3
+      pkgs.alsaLib
+      pkgs.xorg.libXtst
+      pkgs.xorg.libX11
+    ];
```
Comment (Atemu) [inline pkgs/by-name/fi/firefoxpwa/package.nix:114]: Perhaps use firefox-unwrapped's buildInputs?

```suggestion
    targetPkgs = _: firefox-unwrapped.buildInputs;
```

**Comment by pasqui23** [comment]
Apparently the problem was with stale `.local/share/firefoxpwa` and deleting it should fix the issue

**Comment by Atemu** [comment]
Good to hear. This avoids quite a bit of unnecessary complexity :)

**Pull request closed without merging**